File: kora-java-s3-client-minio/src/main/java/ru/tinkoff/kora/example/s3/minio/RootService.java
package ru.tinkoff.kora.example.s3.minio;

import ru.tinkoff.kora.common.Component;
import ru.tinkoff.kora.common.annotation.Root;

@Root
@Component
public class RootService {

    private final SyncS3Client syncS3Client;
    private final AsyncS3Client asyncS3Client;

    public RootService(SyncS3Client syncS3Client, AsyncS3Client asyncS3Client) {
        this.syncS3Client = syncS3Client;
        this.asyncS3Client = asyncS3Client;
    }
}

File: examples/kotlin/kora-kotlin-s3-client-minio/src/main/kotlin/ru/tinkoff/kora/kotlin/example/s3/minio/RootService.kt
package ru.tinkoff.kora.kotlin.example.s3.minio

import ru.tinkoff.kora.common.Component
import ru.tinkoff.kora.common.annotation.Root
import ru.tinkoff.kora.s3.client.annotation.S3
import ru.tinkoff.kora.s3.client.model.*

@Root
@Component
class RootService(private val syncS3Client: SyncS3Client)

